apps/web/ee/components/apiKeys/ApiKeyDialogForm.tsx

```diff
@@ -0,0 +1,163 @@
+  };
+  const {
+    defaultValues = {
+      note: "" as string | undefined,
+      expiresAt: dayjs().add(1, "month").toDate(),
+    } as Omit<TApiKeys, "userId" | "createdAt" | "lastUsedAt">,
+  } = props;
+
+  const [selectedDate, setSelectedDate] = useState(dayjs().add(1, "month").toDate());
+  const [newApiKey, setNewApiKey] = useState("");
+  const [successfulNewApiKeyModal, setSuccessfulNewApiKeyModal] = useState(false);
+  const [newApiKeyDetails, setNewApiKeyDetails] = useState({
+    id: "",
+    hashedKey: "",
+    expiresAt: null as Date | null,
+    note: "" as string | null,
+  });
+
+  const handleDateChange = (e: Date) => {
+    setSelectedDate(e);
+    form.setValue("expiresAt", e);
+  };
+
+  const form = useForm({
+    defaultValues,
+  });
+  return (
+    <>
+      {successfulNewApiKeyModal ? (
+        <>
+          <div className="mb-10">
+            <h2 className="font-semi-bold font-cal mb-2 text-xl tracking-wide text-gray-900">
+              {" "}
+              {t("success_api_key_created")}
+            </h2>
+            <div className="text-sm text-gray-900">
+              <span className="font-semibold">{t("success_api_key_created_bold_tagline")}</span>{" "}
+              {t("you_will_only_view_it_once")}
+            </div>
+          </div>
+          <div>
+            <div className="flex">
+              <code className="my-2 mr-1 w-full truncate rounded-sm bg-gray-100 py-2 px-3 align-middle font-mono text-gray-800">
+                {newApiKey}
+              </code>
+              <Tooltip content={t("copy_to_clipboard")}>
+                <Button
+                  onClick={() => {
+                    navigator.clipboard.writeText(newApiKey);
+                    showToast(t("api_key_copied"), "success");
+                  }}
+                  type="button"
+                  className=" my-2 px-4 text-base">
+                  <ClipboardCopyIcon className="mr-2 h-5 w-5 text-neutral-100" />
+                  {t("copy")}
+                </Button>
+              </Tooltip>
+            </div>
+            <span className="text-sm text-gray-400">
+              {" "}
+              {t("expires")} {newApiKeyDetails?.expiresAt?.toLocaleDateString()}
+            </span>
+          </div>
+          <DialogFooter>
+            <Button type="button" color="secondary" onClick={props.handleClose} tabIndex={-1}>
+              {t("done")}
+            </Button>
+          </DialogFooter>
+        </>
+      ) : (
+        <Form
+          form={form}
+          handleSubmit={async (event) => {
+            try {
+              const newApiKey = await utils.client.mutation("viewer.apiKeys.create", event);
+              setNewApiKey(newApiKey);
+              setNewApiKeyDetails({ ...event });
+              await utils.invalidateQueries(["viewer.apiKeys.list"]);
+              setSuccessfulNewApiKeyModal(true);
+            } catch (error: any) {
+              console.log(error);
```

Comment:
Removed it for now, we need to handle the case of a possible error just if an apikey is not unique for example or whatever. Where would you put that. Maybe move it to check for event (result from mutation) and return error from there? Could show a different error modal in case of error or a toast.